arcgis-ios-sdk-samples/Utility Network/Configure subnetwork trace/ConfigureSubnetworkTraceViewController.swift

```diff
@@ -0,0 +1,389 @@
+            if selectedAttribute != nil {
+                if let domain = selectedAttribute?.domain as? AGSCodedValueDomain {
+                    // Get the value labels.
+                    if valueLabels.isEmpty {
+                        domain.codedValues.forEach { (codedValue) in
+                            valueLabels.append(codedValue.name)
+                        }
+                    }
+                    // Prompt value selection.
+                    let optionsViewController = OptionsTableViewController(labels: valueLabels, selectedIndex: domain.codedValues.count) { (index) in
+                        self.valueLabel?.text = self.valueLabels[index]
+                        self.selectedValue = domain.codedValues[index]
+                    }
+                    optionsViewController.title = "Value"
+                    show(optionsViewController, sender: self)
+                } else {
+                    // Prompt the user to create a custom value if none are available to select.
+                    addCustomValue()
+                }
+            } else {
+                // If no attribute is selected, present an alert.
+                presentAlert(title: "No attribute selected", message: "Please select an attribute before selecting a value.")
+            }
+        } else if cell == addConditionButton {
+            addCondition()
+        } else if cell == resetButton {
+            reset()
+        } else if cell == traceButton {
+            trace()
+        }
+    }
+    
+    func addCustomValue() {
+        // Create an alert controller.
+        let alert = UIAlertController(title: "Create a value", message: "This attribute has no values. Please create one.", preferredStyle: .alert)
+        // Add the text field.
+        alert.addTextField { (textField) in
+            textField.placeholder = "Add a custom value"
+            textField.keyboardType = .numberPad
+        }
+        // Obtain user value.
+        alert.addAction(UIAlertAction(title: "Done", style: .default) { [weak self] _ in
+            let textfield = alert.textFields![0]
+            // Check for invalid input characters.
+            if !(CharacterSet(charactersIn: "0123456789").isSuperset(of: CharacterSet(charactersIn: textfield.text!))) {
+                // Present alert to explain error.
+                self?.presentAlert(title: "This field accepts only numeric entries.")
+            } else {
+                guard let self = self else { return }
+                self.valueLabel?.text = textfield.text
+                self.tableView.reloadRows(at: [IndexPath(row: 3, section: 1)], with: .none)
+                self.selectedValue = textfield.text
+            }
+        })
+        // Add cancel button.
+        alert.addAction(UIAlertAction(title: "Cancel", style: .cancel))
+        
+        // Present the alert.
+        self.present(alert, animated: true, completion: nil)
+    }
+    
+    func addCondition() {
```

Comment:
I got a crash when I didn't actually select anything:

![image](https://user-images.githubusercontent.com/2257493/85480571-19fc7d00-b575-11ea-937e-f4e0ef7c7ce6.png)

It would be best to only enable the "Add Condition" button when there's enough information to create a condition.